mlir/lib/Mitigation/Transforms/MitigationMethods/Zne.cpp

```diff
@@ -246,7 +246,16 @@ FlatSymbolRefAttr globalFolding(Location loc, PatternRewriter &rewriter, std::st
     Value c1 = rewriter.create<index::ConstantOp>(loc, 1);
     TypedAttr numberQubitsAttr = rewriter.getI64IntegerAttr(numberQubits);
     Value numberQubitsValue = rewriter.create<arith::ConstantOp>(loc, numberQubitsAttr);
-    rewriter.create<quantum::DeviceInitOp>(loc, lib, name, kwargs);
+
+    // TODO: in the frontend, calculation of shots will happen outside of the qnode,
+    // before qml.device(..., shots = <some value computed earlier>) is called,
+    // so the SSA def-use computation chain of the shots will actually not be inside the qnode
+    // For now, we simply create a single arith.constant SSA shots value for the ZNE tests.
+    // Revisit when discussing the frontend design of dynamic shots/device/qnode interaction.
+    Operation *shotsLocal = shots->clone();
```

Comment:
Do you have to test dynamic number of shots with grad? (Not sure if we support grad with shots). Whenever I change something in qnode, I usually need to change ZNE, grad, vjp, jvp, etc...

Reply:
Yeah, never mind, I don't think we support taking the gradient of a circuit with samples or counts.

Reply:
Now that you mentioned it, I realized that taking the derivative of sample/counts makes no sense in the first place. It's like taking the derivative of a function that flips a bunch of coins and returns the number of heads....